client-app/shared/wishlists/components/add-to-wishlists-dialog.vue

```diff
@@ -73,44 +138,131 @@ const props = defineProps({
+
+const creationButtonDisabled = computed(() => lists.value.length + inputs.value.length >= listsLimit);
+
+const listsContain = computed(() => {
+  return lists.value.filter((list) => list.items!.some((item) => item.productId === productId));
+});
 
-const extendedLists = computed(() => {
-  const productId = props.product.id;
+function listsRemoveUpdate(id: string, checked: boolean) {
+  const index = listsRemove.value.indexOf(id);
 
-  return lists.value.map((list) => ({
-    ...list,
-    contains: list.items!.some((item) => item.productId === productId),
-  }));
+  if (!checked && index === -1) {
+    listsRemove.value.push(id);
+  } else {
+    listsRemove.value.splice(index, 1);
+  }
+}
+
+const listsOther = computed(() => {
+  return lists.value.filter((list) => !list.items!.some((item) => item.productId === productId));
 });
 
-async function addToWishlists() {
-  if (!selectedListsIds.value.length) {
+function addInput() {
+  inputs.value.push({
+    listName: `${t("shared.wishlists.add_to_wishlists_dialog.new_list")} ${moment().format("YYYY-MM-DD • hh:mm")}`,
+    errorMessage: "",
+  });
+}
+
+function removeInput(index: number) {
+  inputs.value.splice(index, 1);
+}
+
+async function addToWishlistsFromListOther() {
+  if (!selectedListsOtherIds.value.length) {
+    return;
+  }
+
+  await addItemsToWishlists(selectedListsOtherIds.value.map((listId) => ({ listId, productId })));
+}
+
+async function createListsAndAddProduct() {
+  if (!inputs.value.length) {
     return;
   }
 
-  const productId = props.product.id;
+  inputs.value.forEach(async (input) => {
+    await createWishlistAndAddProduct(input.listName, productId);
+  });
+
+  inputs.value.splice(0);
+}
+
+async function removeProductFromWishlists() {
+  const payload = listsRemove.value.map((listId) => {
+    const list = listsContain.value.find((item) => item.id === listId);
+    const lineItemId = list?.items?.find((item) => item.productId === productId)?.id || "";
+    return {
+      listId,
+      lineItemId,
+    };
+  });
+
+  if (payload.length) {
+    await removeItemsFromWishlists(payload);
+  }
+}
+
+async function save() {
+  inputs.value.forEach((input) => {
+    if (!input.listName.trim().length) {
+      input.errorMessage = t("shared.wishlists.add_to_wishlists_dialog.is_required_validation_error");
+    } else {
+      input.errorMessage = "";
+    }
+  });
+  if (inputs.value.filter((input) => !!input.errorMessage?.length).length) {
+    return;
+  }
 
   loading.value = true;
 
-  await addItemsToWishlists(selectedListsIds.value.map((listId) => ({ listId, productId })));
+  await createListsAndAddProduct();
+  await removeProductFromWishlists();
+  await addToWishlistsFromListOther();
 
   closePopup();
-
   loading.value = false;
 
-  openPopup({
-    component: AddedToWishlistsDialog,
-    props: {
-      product: props.product,
-      listIds: selectedListsIds.value,
-    },
+  notifications.success({
+    duration: 15000,
```

Comment:
Looks like this should be moved to core/constants - this interval should be tuned from the single place.

Reply:
done